### ISSUE TITLE
Overloaded functions not recognized

### ISSUE THREAD

**Issue by yanxke** [microsoft/pylance-release#4526]
Red squiggles under x, y even though it matches one of the valid function definitions.

![image](https://github.com/microsoft/pylance-release/assets/25883626/dbd33144-0d51-47e0-9719-6e1a3b605919)

## Environment data

-   Language Server version: v2023.6.20
-   OS and version: Windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.6

**Comment by erictraut** [comment]
Please provide a minimal self-contained code sample that demonstrates the issue. It's unlikely that we can repro this or diagnose the problem from a partial screen shot.

**Comment by yanxke** [comment]
The repro is just that one line.

![image](https://github.com/microsoft/pylance-release/assets/25883626/d7238a2a-b643-430a-8fa8-a7d85dac9698)

**Comment by debonte** [comment]
```python
import cv2

image = cv2.imread('src/lena.jpg')
x = 0
y = 0
width = 100
height = 100
cv2.rectangle(image, (x, y), (x + width, y + height), (0, 0, 255), 2)
```

**Comment by erictraut** [comment]
Thanks. For future bug reports, a text version would be appreciated so we don't need to retype your code from a screen shot.

```python
import cv2

image = cv2.imread("")
x = 0
y = 0
width = 100
height = 100
cv2.rectangle(image, (x, y), (x + width, y + height), (0, 0, 255), 2)
```

There is a bug in the `cv2` stub that ship with pylance. It defines `rectangle` as an overload using the `overload` decorator, but it fails to import `overload` from the typing module. If you replace `@overload` with `@typing.overload`, it works fine.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.6.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023641-28-June-2023-PreRelease)
